dm/tests/shardddl4_1/run.sh

```diff
@@ -823,11 +823,91 @@ function DM_152 {
 	run_case 152 "double-source-optimistic" "init_table 111 211 212" "clean_table" "optimistic"
 }
 
+function DM_153_CASE {
+	run_sql_source1 "insert into ${shardddl1}.${tb1} values(1,1);"
+	run_sql_source2 "insert into ${shardddl1}.${tb1} values(2,2);"
+
+	run_sql_source1 "alter table ${shardddl1}.${tb1} add column b int, drop column c;"
+	run_sql_source1 "insert into ${shardddl1}.${tb1} values(4,4);"
+	run_sql_source2 "insert into ${shardddl1}.${tb1} values(5,5);"
+
+	run_sql_source1 "alter table ${shardddl1}.${tb1} add column c int, drop column b;"
+	run_sql_source1 "insert into ${shardddl1}.${tb1} values(7,7);"
+	run_sql_source2 "insert into ${shardddl1}.${tb1} values(8,8);"
+
+	run_dm_ctl_with_retry $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+		"query-status test" \
+		"because schema conflict detected" 1 \
+		"add column c that wasn't fully dropped in downstream" 1
+
+	# try to fix data
+	run_dm_ctl $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+		"binlog-schema update test ${shardddl1} ${tb1} -s mysql-replica-01 --from-target" \
+		"\"result\": true" 2
+
+	run_dm_ctl_with_retry $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+		"binlog replace test \"alter table ${shardddl1}.${tb1} drop column b\"" \
+		"\"result\": true" 2 \
+		"\"source 'mysql-replica-02' has no error\"" 1
+
```

Comment:
do we need use resume-task to check if this err is already be handeld?

Reply:
i think if `check_sync_diff` success, the errors about ddls is handled, because there still have some dml after ddls.

Reply:
i see, thanks